c10/core/impl/PyObjectSlot.cpp

```diff
@@ -47,6 +51,15 @@ PyInterpreter& PyObjectSlot::load_pyobj_interpreter() const {
       (*pyobj_interpreter_.load())->name());
 }
 
+bool PyObjectSlot::check_interpreter(PyInterpreter* interpreter) {
+  return interpreter == pyobj_interpreter();
+}
+
+bool PyObjectSlot::has_pyobj() {
```

Comment:
It seems important to note in the method name here that you MUST NOT be hermetic when you invoke this, which appears to be true at the sole call site